api/institutions/views.py

```diff
@@ -0,0 +1,247 @@
+
+    All institutions are available to be read by everyone. However, no one has write
+    permissions for institutions.
+
+    ##Attributes
+
+    OSF Institutions have the "institutions" `type`.
+
+        name           type               description
+        -------------------------------------------------------------------------
+        name           string             title of the institution
+        id             string             unique identifier in the OSF
+        logo_path      string             a path to the institution's static logo
+
+    ##Relationships
+
+    ###Nodes
+    List of nodes that have this institution as its primary institution.
+
+    ###Users
+    List of users that are affiliated with this institution.
+
+    ##Links
+
+        self:  the canonical api endpoint of this node
+        html:  this node's page on the OSF website
+
+    #This Request/Response
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.INSTITUTION_READ]
+    required_write_scopes = [CoreScopes.NULL]
+    model_class = Institution
+
+    serializer_class = InstitutionSerializer
+    view_category = 'institutions'
+    view_name = 'institution-detail'
+
+    # overrides RetrieveAPIView
+    def get_object(self):
+        return self.get_institution()
+
+
+class InstitutionNodeList(JSONAPIBaseView, ODMFilterMixin, generics.ListAPIView, InstitutionMixin):
+    """Nodes that have selected an institution as their primary institution.
+
+    ##Permissions
+    Only public nodes or ones in which current user is a contributor.
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.INSTITUTION_READ, CoreScopes.NODE_BASE_READ]
+    required_write_scopes = [CoreScopes.NULL]
+    model_class = Node
+
+    serializer_class = NodeSerializer
+    view_category = 'institutions'
+    view_name = 'institution-nodes'
+
+    # overrides ODMFilterMixin
+    def get_default_odm_query(self):
+        inst = self.get_institution()
+        inst_query = Q('primary_institution', 'eq', inst)
+        base_query = (
+            Q('is_deleted', 'ne', True) &
+            Q('is_folder', 'ne', True) &
+            Q('is_registration', 'eq', False)
+        )
+        user = self.request.user
+        permission_query = Q('is_public', 'eq', True)
+        if not user.is_anonymous():
+            permission_query = (permission_query | Q('contributors', 'icontains', user._id))
+
+        query = base_query & permission_query & inst_query
+        return query
+
+    # overrides RetrieveAPIView
+    def get_queryset(self):
+        query = self.get_query_from_request()
+        return Node.find(query)
+
+
+class InstitutionNodeDetail(JSONAPIBaseView, generics.RetrieveAPIView, NodeMixin, InstitutionMixin):
```

Comment:
Related to this comment: https://github.com/CenterForOpenScience/osf.io/pull/4661/files#r47368430

Is this a needed view?